CorePayments/src/main/java/com/paypal/android/corepayments/api/PatchCCOWithAppSwitchEligibility.kt

```diff
@@ -0,0 +1,76 @@
+package com.paypal.android.corepayments.api
+
+import android.content.Context
+import com.paypal.android.corepayments.APIClientError
+import com.paypal.android.corepayments.CoreConfig
+import com.paypal.android.corepayments.PayPalSDKError
+import com.paypal.android.corepayments.RestClient
+import com.paypal.android.corepayments.common.Headers
+import com.paypal.android.corepayments.graphql.GraphQLClient
+import com.paypal.android.corepayments.graphql.GraphQLResult
+import com.paypal.android.corepayments.model.APIResult
+import com.paypal.android.corepayments.model.ExperimentationContext
+import com.paypal.android.corepayments.model.ExternalResponse
+import com.paypal.android.corepayments.model.PatchCcoWithAppSwitchEligibilityRequest
+import com.paypal.android.corepayments.model.PatchCcoWithAppSwitchEligibilityResponse
+import com.paypal.android.corepayments.model.TokenType
+import com.paypal.android.corepayments.model.Variables
+
+class PatchCCOWithAppSwitchEligibility internal constructor(
+    private val authenticationSecureTokenServiceAPI: AuthenticationSecureTokenServiceAPI,
+    private val graphQLClient: GraphQLClient,
+) {
+
+    constructor(coreConfig: CoreConfig) : this(
+        authenticationSecureTokenServiceAPI = AuthenticationSecureTokenServiceAPI(
+            coreConfig,
+            RestClient(coreConfig)
+        ),
+        graphQLClient = GraphQLClient(coreConfig)
+    )
+
+    suspend operator fun invoke(
+        context: Context,
+        orderId: String,
+        tokenType: TokenType,
+        merchantOptInForAppSwitch: Boolean,
+        paypalNativeAppInstalled: Boolean
+    ): APIResult<PatchCcoWithAppSwitchEligibilityResponse> {
+        return runCatching {
+            val token =
+                when (val tokenResult = authenticationSecureTokenServiceAPI.createLowScopedAccessToken()) {
+                    is APIResult.Success -> tokenResult.data
+                    is APIResult.Failure -> return APIResult.Failure(tokenResult.error)
```

Comment:
🧶 : Is there a way to clean up this return statement? I understand what it's doing, but it seems sort of like an escape hatch. It might read a lot cleaner if we make the output of `when` be the return value of the `runCatching` block, rather than assigning the result of `when(){}` to `token`.